lib/routes/pos/transactions/pos_transactions_page.dart

```diff
@@ -55,73 +55,103 @@ class _PosTransactionsState extends State<_PosTransactionsPage> {
                   stream: widget._accountBloc.paymentsStream,
                   builder: (context, snapshot) {
-                    List<PaymentInfo> payments = new List<PaymentInfo>();
+                    PaymentsModel paymentsModel = new PaymentsModel.initial();
                     if (snapshot.hasData) {
-                      payments = snapshot.data.paymentsList ?? new List<PaymentInfo>();
-                      _firstDate = snapshot.data.firstDate ?? DateTime(2018);
+                      paymentsModel = snapshot.data;
                     }
 
-                    if (account == null || payments == null) {
+                    if (account == null || paymentsModel == null) {
                       // build loading page, waiting for account to initialize || this is temporary
-                      return Center(child: Loader());
+                      return _buildScaffold(Center(child: Loader()));
                     }
 
-                    if (account.balance == 0 && payments.length == 0) {
-                      // build empty account page || this is temporary
-                      return Center(child: Text("Successful transactions are displayed here."),);
+                    if (account.balance == 0 && paymentsModel.paymentsList.length == 0) {
+                      return _buildScaffold(Center(child: Text("Successful transactions are displayed here."),), _calendarButton(paymentsModel.firstDate));
                     }
 
                     // account and payments are ready, build their widgets
-                    return _buildTransactions(payments);
+                    return _buildScaffold(_buildTransactions(paymentsModel),_calendarButton(paymentsModel.firstDate));
                   });
-            }));
+            });
+  }
+
+  Widget _buildScaffold(Widget body, [Widget actions]) {
+    return new Scaffold(
+      key: _scaffoldKey,
+      appBar: new AppBar(
+        iconTheme: theme.appBarIconTheme,
+        textTheme: theme.appBarTextTheme,
+        backgroundColor: Color.fromRGBO(5, 93, 235, 1.0),
+        leading: backBtn.BackButton(),
+        title: new Text(
+          _title,
+          style: theme.appBarTextStyle,
+        ),
+        actions: <Widget>[actions == null ? Container() : actions],
+        elevation: 0.0,
+      ),
+      body: body,
+    );
   }
 
-  Widget _buildTransactions(List<PaymentInfo> payments) {
+  Widget _calendarButton(DateTime firstDate) {
+    return Padding(padding: EdgeInsets.only(right: 16.0), child:
+    IconButton(icon: ImageIcon(AssetImage("src/icon/calendar.png"), color: Colors.white, size: 24.0),
+        onPressed: () =>
+            showDialog(
+                context: context,
+                builder: (_) => CalendarDialog(context, firstDate)).then(((result) =>
+                widget._accountBloc.paymentFilterSink.add(PaymentFilterModel(
+                    [PaymentType.RECEIVED, PaymentType.DEPOSIT, PaymentType.SENT, PaymentType.WITHDRAWAL
```

Comment:
Here you reset the payment type filter, don't you want to take the current filter and just use copyWith function with the dates?